ext/systemd-daemon/systemd-daemon.c

```diff
@@ -16,6 +16,54 @@ static VALUE _sd_notify(VALUE mod, VALUE unset_env, VALUE state)
   return INT2FIX(return_code);
 }
 
+/* Supports only one fd per call */
+static VALUE _sd_pid_notify_with_fds(VALUE mod, VALUE pid, VALUE unset_env, VALUE state, VALUE fds, VALUE fds_size)
+{
+  int i, return_code;
+
+#ifdef HAVE_SD_PID_NOTIFY_WITH_FDS
+  const char *sd_state;
+  int size = FIX2INT(fds_size);
+  int *fds_ref = (int *)malloc(size * sizeof(int));
+
+  sd_state = StringValuePtr(state);
+  for(i = 0; i < size; i ++)
+    fds_ref[i] = FIX2INT(rb_ary_entry(fds, i));
+  return_code = sd_pid_notify_with_fds(FIX2INT(pid), FIX2INT(unset_env), sd_state, fds_ref, size);
```

Comment:
You can use `NUM2PIDT()` here that returns a `pid_t`. Before that, you can guard it using `FIXNUM_P()` for type checking.